genie-client/src/main/java/com/netflix/genie/client/interceptors/ResponseMappingInterceptor.java

```diff
@@ -66,6 +68,9 @@ public Response intercept(final Chain chain) throws IOException {
                             responseBody == null || !responseBody.has(ERROR_MESSAGE_KEY)
                                 ? NO_MESSAGE_FALLBACK
                                 : responseBody.get(ERROR_MESSAGE_KEY).asText();
+                        if (response.code() == HTTP_TOO_MANY_REQUESTS) {
```

Comment:
nit: There's now 2 possible calls to `response.code()` here and the line below in the original throw. Up to you if you want to break it out.

Reply:
Done.